packages/sit-onyx/src/components/OnyxBadge/OnyxBadge.vue

```diff
@@ -0,0 +1,110 @@
+<script lang="ts" setup>
+import { useDensity } from "../../composables/density";
+import OnyxIcon from "../OnyxIcon/OnyxIcon.vue";
+import type { OnyxBadgeProps } from "./types";
+
+const props = withDefaults(defineProps<OnyxBadgeProps>(), {
+  variation: "primary",
+});
+
+const { densityClass } = useDensity(props);
+
+defineSlots<{
+  /**
+   * Badge content.
+   */
+  default?(): unknown;
+}>();
+</script>
+
+<template>
+  <div
+    class="onyx-badge"
+    :class="[
+      'onyx-truncation-ellipsis',
+      'onyx-text',
+      `onyx-badge--${props.variation}`,
+      densityClass,
+    ]"
+  >
+    <OnyxIcon
+      v-if="props.icon"
+      class="onyx-badge__icon"
+      :icon="props.icon"
+      :size="props.density === 'compact' ? '16px' : '24px'"
+    />
+    <slot v-else></slot>
+  </div>
+</template>
+
+<style lang="scss">
+@use "../../styles/density.scss";
+
+.onyx-badge {
+  @include density.compact {
+    --onyx-badge-padding: 0 var(--onyx-spacing-2xs);
+    --onyx-badge-icon-padding: var(--onyx-spacing-4xs);
+    --onyx-badge-gap: var(--onyx-spacing-4xs);
+    --onyx-badge-height: 1.5rem;
+  }
+
+  @include density.default {
+    --onyx-badge-padding: var(--onyx-spacing-5xs) var(--onyx-spacing-sm);
+    --onyx-badge-icon-padding: var(--onyx-spacing-5xs);
+    --onyx-badge-gap: var(--onyx-spacing-2xs);
+    --onyx-badge-height: 1.75rem;
+  }
+
+  @include density.cozy {
+    --onyx-badge-padding: var(--onyx-spacing-4xs) var(--onyx-spacing-sm);
+    --onyx-badge-icon-padding: var(--onyx-spacing-4xs);
+    --onyx-badge-gap: var(--onyx-spacing-sm);
+    --onyx-badge-height: 2rem;
+  }
+
+  --onyx-badge-background-color: var(--onyx-color-base-primary-500);
+
+  &:has(&__icon) {
+    --onyx-badge-padding: var(--onyx-badge-icon-padding);
+  }
+
+  display: inline-block;
+  height: var(--onyx-badge-height);
+  max-width: 100%;
+  box-sizing: border-box;
+  padding: var(--onyx-badge-padding);
+  border-radius: var(--onyx-radius-full);
+  background-color: var(--onyx-badge-background-color);
+  color: var(--onyx-color-text-icons-neutral-inverted);
+  font-family: var(--onyx-font-family);
+  font-style: normal;
+
+  &--secondary {
+    --onyx-badge-background-color: var(--onyx-color-base-neutral-700);
```

Comment:
```suggestion
    --onyx-badge-background-color: var(--onyx-color-base-secondary-500);
```

Currently secondary and neutral are both neutral but secondary is a different color

Reply:
Now both primary and secondary have the same color:
![image](https://github.com/SchwarzIT/onyx/assets/60856270/af1a59a8-a234-4f12-a6ac-8af2bf71e392)


Should we maybe wait for Jannick's decision and then make a change?

Reply:
Thats correct at the moment because our default theme has the same color for both. But the c[olor palette](https://onyx.schwarz/tokens/colors.html) defines both so if you define your own colors/theme they will differ (e.g. blue and red for the Kaufland theme)